azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -653,6 +688,10 @@ func resourceArmVirtualMachineScaleSetRead(d *schema.ResourceData, meta interfac
 		d.Set("extension", extension)
 	}
 
+	if resp.Plan != nil {
+		d.Set("plan", flattenAzureRmVirtualMachineScaleSetPlan(resp.Plan))
```

Comment:
for complex Set operations like this, it would be good to throw an error if the set fails